services/worker-manager/src/api.js

```diff
@@ -839,6 +841,137 @@ builder.declare({
   return res.reply({ expires: expires.toJSON(), credentials, secret: newSecret });
 });
 
+builder.declare({
+  method: 'get',
+  route: '/provisioners/:provisionerId/worker-types/:workerType/workers',
+  query: {
+    ...paginateResults.query,
+    quarantined: /^(true|false)$/,
+    requested: /^(true|false)$/,
+    running: /^(true|false)$/,
+    stopping: /^(true|false)$/,
+    stopped: /^(true|false)$/,
+  },
+  name: 'listWorkers',
+  scopes: 'worker-manager:list-workers:<provisionerId>/<workerType>',
+  stability: APIBuilder.stability.experimental,
+  category: 'Worker Metadata',
+  output: 'list-workers-response.yml',
+  title: 'Get a list of all active workers of a workerType',
+  description: [
+    'Get a list of all active workers of a workerType.',
+    '',
+    '`listWorkers` allows a response to be filtered by quarantined and non quarantined workers,',
+    'as well as the current state of the worker.',
+    'To filter the query, you should call the end-point with one of [`quarantined`, `requested`, `running`,',
+    '`stopping`, `stopped`] as a query-string option with a true or false value.',
+    '',
+    'The response is paged. If this end-point returns a `continuationToken`, you',
+    'should call the end-point again with the `continuationToken` as a query-string',
+    'option. By default this end-point will list up to 1000 workers in a single',
+    'page. You may limit this with the query-string parameter `limit`.',
+  ].join('\n'),
+}, async function(req, res) {
+  const quarantined = req.query.quarantined || null;
+  const requested = req.query.requested || null;
+  const running = req.query.running || null;
+  const stopping = req.query.stopping || null;
+  const stopped = req.query.stopped || null;
+  const workerState = requested || running || stopping || stopped || null;
```

Comment:
Why don't we just use enum-like query parameter? It seems like they are mutually exclusive anyway and only will be filtered by one of the values. So instead of having 5 boolean params, why not have 1 `workerState` of type string/enum? Validation is already done above with regexes.

Reply:
Ok, after checking the component's code below, maybe extra params are fine, we just need to fix line 880, as I'm pretty sure it can also be any of `true|false|null` values so will not be helpful to filter states

Reply:
I'm gonna combine to just one `workerState`. I initially had it written that way and changed my mind for some reason.